private/model/api/load.go

```diff
@@ -7,8 +7,123 @@ import (
 	"fmt"
 	"os"
 	"path/filepath"
+	"sort"
+	"strings"
 )
 
+// APIs provides a set of API models loaded by API package name.
+type APIs map[string]*API
+
+// LoadAPIs loads the API model files from disk returning the map of API
+// package. Returns error if multiple API model resolve to the same package
+// name.
+func LoadAPIs(modelPaths []string, baseImport string) (APIs, error) {
+	apis := APIs{}
+	for _, modelPath := range modelPaths {
+		a, err := loadAPI(modelPath, baseImport)
+		if err != nil {
+			return nil, fmt.Errorf("failed to load API, %v, %v", modelPath, err)
+		}
+		importPath := a.ImportPath()
+		if _, ok := apis[importPath]; ok {
+			return nil, fmt.Errorf(
+				"package names must be unique attempted to load %v twice. Second model file: %v",
+				importPath, modelPath)
+		}
+		apis[importPath] = a
+	}
+
+	return apis, nil
+}
+
+func loadAPI(modelPath, baseImport string) (*API, error) {
+	a := &API{
+		BaseImportPath:   baseImport,
+		BaseCrosslinkURL: "https://docs.aws.amazon.com",
+	}
+
+	err := attachModelFiles(modelPath,
+		modelLoader{"api-2.json", a.Attach, true},
+		modelLoader{"paginators-1.json", a.AttachPaginators, false},
+		modelLoader{"docs-2.json", a.AttachDocs, false},
```

Comment:
Should doc-2.json not be required?

Reply:
Yeah can make it required. Not having the doc model doesn't prevent code generation, but does mean modeled docs which isn't good.